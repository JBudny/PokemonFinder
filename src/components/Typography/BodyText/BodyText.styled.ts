import { Text } from 'react-native'

import styled from 'styled-components/native'

import { TextSize } from '@components/Typography/Typography.types'

export const StyledBodyText = styled(Text)<TextSize>`
	font-size: ${({ size, theme }) =>
		(size === 'large' && `${theme.designSystem.bodyTextLargeSize}px`) ||
		(size === 'medium' && `${theme.designSystem.bodyTextMediumSize}px`) ||
		(size === 'small' && `${theme.designSystem.bodyTextSmallSize}px`)};
	font-weight: ${({ theme }) => theme.designSystem.bodyTextWeight};
	color: ${({ theme }) => theme.palette.text};
`
