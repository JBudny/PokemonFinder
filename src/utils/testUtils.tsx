import React from 'react'

import type { PreloadedState } from '@reduxjs/toolkit'
import type { RenderOptions } from '@testing-library/react-native'
import { render } from '@testing-library/react-native'

import type { AppStore, RootState } from '@src/store'
import { setupStore } from '@src/store'

import { Providers } from './Providers'

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store. For
// future dependencies, such as wanting to test with react-router, you can extend
// this interface to accept a path and route and use those in a <MemoryRouter />
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>
	store?: AppStore
}

const renderWithProviders = (
	ui: React.ReactElement,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {}
) => {
	const Wrapper: React.FC = ({ children }) => (
		<Providers store={store}>{children}</Providers>
	)

	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export { renderWithProviders }
