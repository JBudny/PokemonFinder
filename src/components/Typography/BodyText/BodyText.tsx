import React, { FC } from 'react'

import { TypographyProps } from '@components/Typography/Typography.types'

import { StyledBodyText } from './BodyText.styled'

/**
 * BodyText component
 * @param size
 * @param text
 * EXTENDS: TextProps
 */
export const BodyText: FC<TypographyProps> = ({ size, text, ...props }) => (
	<StyledBodyText size={size} {...props}>
		{text}
	</StyledBodyText>
)
