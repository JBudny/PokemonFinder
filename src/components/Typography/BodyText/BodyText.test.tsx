import React from 'react'

import { renderWithProviders } from '@utils/testUtils'

import { BodyText } from './BodyText'

test('BodyText component should display provided text', () => {
	const { getByText } = renderWithProviders(
		<BodyText size="small" text="test text" />
	)

	expect(getByText('test text')).toBeDefined()
})

test('BodyText component return proper text size 16dp size text', () => {
	const { getByText } = renderWithProviders(
		<BodyText size="small" text="test text" />
	)
	expect(getByText('test text')).toHaveStyle({ fontSize: 16 })
})

test('BodyText component with medium text prop should return 20dp size text', () => {
	const { getByText } = renderWithProviders(
		<BodyText size="medium" text="test text" />
	)
	expect(getByText('test text')).toHaveStyle({ fontSize: 20 })
})

test('BodyText component with large text prop should return 25dp size text', () => {
	const { getByText } = renderWithProviders(
		<BodyText size="large" text="test text" />
	)
	expect(getByText('test text')).toHaveStyle({ fontSize: 25 })
})

test('BodyText component should match snapshot', () => {
	const text = renderWithProviders(<BodyText size="small" text="test text" />)

	expect(text).toMatchSnapshot()
})
