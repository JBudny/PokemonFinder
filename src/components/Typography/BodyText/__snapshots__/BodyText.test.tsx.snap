// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyText component should match snapshot 1`] = `
<Text
  size="small"
  style={
    Array [
      Object {
        "color": "rgb(33, 26, 29)",
        "fontSize": 16,
        "fontWeight": "normal",
      },
    ]
  }
>
  test text
</Text>
`;
